model/project_vars.go

```diff
@@ -66,6 +68,61 @@ func FindOneProjectVars(projectId string) (*ProjectVars, error) {
 	return projectVars, nil
 }
 
+// FindMergedProjectVars merges vars from the target project's ProjectVars and its parent repo's vars
+func FindMergedProjectVars(projectID string) (*ProjectVars, error) {
+	project, err := FindOneProjectRef(projectID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project '%s'", projectID)
+	}
+	if project == nil {
+		return nil, errors.New(fmt.Sprintf("project '%s' does not exist", projectID))
+	}
+
+	projectVars, err := FindOneProjectVars(projectID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project vars for project '%s'", projectID)
+	}
+	if !project.UseRepoSettings {
+		return projectVars, nil
+	}
+
+	repoVars, err := FindOneProjectVars(project.RepoRefId)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project vars for repo '%s'", project.RepoRefId)
+	}
+	if repoVars != nil {
+		if projectVars == nil {
+			projectVars = repoVars
+			projectVars.Id = project.Id
+		} else {
+			if projectVars.Vars == nil {
+				projectVars.Vars = map[string]string{}
+			}
+			if projectVars.PrivateVars == nil {
+				projectVars.PrivateVars = map[string]bool{}
+			}
+			if projectVars.RestrictedVars == nil {
+				projectVars.RestrictedVars = map[string]bool{}
+			}
+
+			// Merge repo vars and project vars
```

Comment:
Can you add to this comment to say that branch-level vars have priority? To explain why you only add to the project vars in the case that it doesn't already exist